test/files/neg/t4425b.check

```diff
@@ -22,22 +22,22 @@ t4425b.scala:10: error: object X is not a case class, nor does it have a valid u
 Note: def unapply(x: String)(y: String): Nothing exists in object X, but it cannot be used as an extractor due to its second non-implicit parameter list
     println((X: Any) match { case X(_, _) => "ok" ; case _ => "fail" })
                                   ^
-t4425b.scala:18: error: The result type of an unapply method must contain a member `get` to be used as an extractor pattern, no such member exists in Nothing
+t4425b.scala:18: error: too many patterns for object X offering Boolean: expected 0, found 2
```

Comment:
Definitely we should keep the previous warning.  I think you can use `isNonBottomSubClass` instead of `<:<`, but maybe you already had a plan.

Reply:
Alright, will do.

Or should it maybe be note that it can't be `Nothing`?

Or should unapply methods returning `Nothing` maybe just work, and non-termninate at runtime? That's the most principled but also probably the least useful thing to do.

Reply:
Hmm.  Maybe it should??  Say I temporarily redefine an extractor with `???` do I expect all my extractor usages to error out?  Probably not.  So maybe you're right that this should work...  But I'm not 100% sure.

Reply:
I'm leaning mostly towards `def unapply = ???` being it's own custom error, something along the lines of "extractors may not return Nothing from their unapply method".

That way, it leads you on the trail of what you should do (e.g. annotate as `Option[Extracted]` while keeping the body `???`).

All other options seem less helpful. I'm warming up just allowing it as an extractor for any arity, but that's probably more work than the effort is worth.

To me, the message that Nothing doesn't have a get method is the yankiest, closely followed by the message this PR makes instead, but in the end it's a pretty exotic use case anyway and a bit of yank gives flavour.

Anyway, your call on what to go for within the scope of this PR.

Reply:
I grokked what it was saying and why, but it wasn't obvious.  The "expected 0, found 2" is misleading IMO.  The reference to Nothing in the previous makes it a better state IMO.  But I do agree it's an exotic area.

Hmm, I'll let it stew for a couple of days.  See if anyone else wants to chime in.  Otherwise, I'm really happy to get these fixes in, thank you!!

Reply:
updated to "object X can't be used as an extractor: The result type of an unapply method may not be Nothing"